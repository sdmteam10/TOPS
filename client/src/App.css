

.remove-btn {
    margin-right: 0.5rem;
}

.fade-enter {
    opacity: 0.01;
}

.fade-enter-active {
    opacity: 1;
    transition: opacity 1000ms ease-in;
}

.fade-exit {
    opacity: 1
}

.fade-exit-active {
    opacity: 0.01;
    transition: opacity 1000ms ease-in;
}

.accordion {
    background-color: #eee;
    color: #444;
    cursor: pointer;
    padding: 18px;
    width: 100%;
    text-align: left;
    border: none;
    outline: none;
    transition: 0.4s;
}

.active, .accordion:hover {
    background-color: #ccc;
}

.panel {
    padding: 0 18px;
    background-color: white;
    max-height: 0;
    overflow: hidden;
    transition: max-height 0.2s ease-out;
}

.accordion:after {
    content: '\002B';  
    color: #777;
    font-weight: bold;
    float: right;
    margin-left: 5px;
}

.active:after { 
    content: "\2212";  
}

.bottom-alert {
    z-index: 10;
    position: fixed;
    width: 100%;
    height:0px;
    left:50%;
    transform:translateX(-50%);
    bottom:80px;
}

