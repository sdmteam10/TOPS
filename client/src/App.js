import './App.css'
import 'bootstrap/dist/css/bootstrap.min.css'
import 'bootstrap/dist/js/bootstrap.bundle'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import Home from './components/Home'
import Onboarder from './components/Onboarder'
import Designer from './components/Designer'
import GoalsList from './components/GoalsList'
import AppNavBar from './components/AppNavBar'
import TitleActivity from './components/TitleActivity'
import TitleOrgInfo from './components/TitleOrgInfo';
import OrgInfo from './components/OrgInfo'
import TitleHome from './components/TitleHome';


function App() {
  let pathname = window.location.pathname
  let titleDesc

  if (pathname === '/Goals' ) {
    titleDesc = <TitleActivity />
  } else if (pathname === '/OrgInfo' ) {
    titleDesc = <TitleOrgInfo />
  } else if (pathname === '/Home' || pathname === '/')
    titleDesc = <TitleHome />
  

  return (
    
    <>
      <AppNavBar />
      {titleDesc}
      <div className='container-md' style={{height: '500px'}}>
        {/* 路由入口 */}
        <Router>
          <Switch>
            {/* 路由规则 */}
            <Route path="/" component={Home} exact />
            <Route path="/Home" component={Home} />
            <Route path="/Onboarder" component={Onboarder} />
            <Route path="/Designer" component={Designer} />
            <Route path="/Goals" component={GoalsList} />
            <Route path="/OrgInfo" component={OrgInfo} />
            {/* <Route path="/Designer" render={() =>
              <Designer>
                <Route exat path="/Designer" component={Designer} />
                <Route path="Goals" component={GoalsList} />
                {/* <Route path="/Designer/OrgInfo" component={GoalsList} /> */}
              {/* </Designer>} */}
          </Switch>
        </Router>
      </div>
    </>
  )}

export default App;

