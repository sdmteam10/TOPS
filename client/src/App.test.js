import * as React from 'react'
import App from './App'
import renderer from 'react-test-renderer';

import "regenerator-runtime/runtime.js";
const regeneratorRuntime = require("regenerator-runtime");

test('Create a snapshot of App', () => {
  const component = renderer.create(
    <App />
  );
  let testSnap = component.toJSON();
  
  expect(testSnap).toMatchSnapshot();
});