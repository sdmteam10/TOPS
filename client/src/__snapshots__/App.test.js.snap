// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a snapshot of App 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary font-monospace flex-row"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand d-flex w-0 me-0"
        href="/Home"
      >
        TOPS
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse w-100"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav nav-fill w-50 mx-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/Onboarder"
            >
              Onboarder
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/Designer"
            >
              Designer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/Contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="bg-primary font-monospace"
    style={
      Object {
        "height": "auto",
      }
    }
  >
    <div
      className="text-light mx-auto text-center"
    >
      <br />
      <h2>
        Team Onboarding Process Support
      </h2>
      <br />
      <h6>
         Designing an onboarding program that is personalised 
      </h6>
      <h6>
         to the characteristics of the onboarder and 
      </h6>
      <h6>
         the team and the organisational context. 
      </h6>
      <br />
    </div>
  </div>,
  <div
    className="container-md"
    style={
      Object {
        "height": "500px",
      }
    }
  >
    <div
      className="mt-md-5 ws-0 mx-md-auto"
    >
      <div
        className="font-monospace"
      >
        Home Page
      </div>
    </div>
  </div>,
]
`;
